sigridci/sigridci/repository_history_exporter.py

```diff
@@ -27,16 +31,35 @@ def exportHistory(self, sourceDir):
             UploadLog.log("No repository history found")
 
     def exportGitHistory(self, sourceDir):
-        gitCommand = ["git", "-C", sourceDir, "--no-pager", "log", "--date=iso",
-                      "--format='@@@;%H;%an;%ae;%ad;%s'", "--numstat", "--no-merges"]
+        gitCommand = ["git", "-C", sourceDir, "--no-pager", "log", "--date=iso", f"--format='{self.GIT_LOG_FORMAT}'",
+                      "--numstat", "--no-merges", f"--after={self.CUTOFF_DATE.strftime('%Y-%m-%d')}"]
 
         try:
             output = subprocess.run(gitCommand, stdout=subprocess.PIPE)
             if output.returncode == 0:
                 UploadLog.log("Including repository history in upload")
-                with open(f"{sourceDir}/git.log", "w") as f:
-                    f.write(output.stdout.decode("utf8"))
+                history = output.stdout.decode("utf8")
+                self.createHistoryExportFile(history, f"{sourceDir}/git.log")
             else:
                 UploadLog.log("Exporting repository history failed")
         except Exception as e:
             UploadLog.log("Error while trying to include repository history: " + str(e))
+
+    def createHistoryExportFile(self, history, outputFile):
+        with open(outputFile, "w") as f:
+            for line in history.strip().split("\n"):
+                f.write(self.anonymizeHistoryEntry(line))
+
+    def anonymizeHistoryEntry(self, gitLog):
+        anonymized = ""
+
+        for line in gitLog.strip().split("\n"):
+            if line.startswith(("@@@", "'@@@")):
```

Comment:
Is there a way to make the code inside this block more self-explanatory? In its current form, it is clear it is about anonymization, but not about _what_ we're anonymizing. The only thing I could think of would be to introduce "unnecessary" variables, something like:
```python
author = cell[2]
committer = cell[3]
```
But that's just me guessing those cells are actually the author and committer, respectively.

Reply:
Python has a somewhat obscure syntax for this. Would you be OK with the following:

```
aap = ["a", "b", "c"]
a, b, c = aap.split(",")
```

Reply:
Obscure it is, but it makes it much clearer what we're doing!